cpp/ql/src/Architecture/Refactoring Opportunities/ClassesWithManyFields.ql

```diff
@@ -20,66 +23,143 @@ string kindstr(Class c)
   )
 }
 
+/**
+ * Holds if the arguments correspond to information about a `VariableDeclarationEntry`.
+ */
 predicate vdeInfo(VariableDeclarationEntry vde, Class c, File f, int line)
 {
   c = vde.getVariable().getDeclaringType() and
   f = vde.getLocation().getFile() and
   line = vde.getLocation().getStartLine()
 }
 
-predicate previousVde(VariableDeclarationEntry previous, VariableDeclarationEntry vde)
+newtype TVariableDeclarationInfo =
+  TVariableDeclarationLine(Class c, File f, int line) {
+  	vdeInfo(_, c, f, line)
+  }
+
+/**
+ * A line that contains one or more `VariableDeclarationEntry`s (in the same class).
+ */
+class VariableDeclarationLine extends TVariableDeclarationInfo
 {
-  exists(Class c, File f, int line | vdeInfo(vde, c, f, line) |
-    vdeInfo(previous, c, f, line - 3) or
-    vdeInfo(previous, c, f, line - 2) or
-    vdeInfo(previous, c, f, line - 1) or
-    (vdeInfo(previous, c, f, line) and exists(int prevCol, int vdeCol |
-      prevCol = previous.getLocation().getStartColumn() and vdeCol = vde.getLocation().getStartColumn() |
-      prevCol < vdeCol or (prevCol = vdeCol and previous.getName() < vde.getName())
-    ))
-  )
+  Class c;
+  File f;
+  int line;
+
+  VariableDeclarationLine() {
+    vdeInfo(_, c, f, line) and
+    this = TVariableDeclarationLine(c, f, line)
+  }
+
+  /**
+   * Gets the class associated with this `VariableDeclarationLine`.
+   */
+  Class getClass() {
+    result = c
+  }
+
+  /**
+   * Gets the line of this `VariableDeclarationLine`.
+   */
+  int getLine() {
+    result = line
+  }
```

Comment:
Please auto-format this query. I'm guessing that'll be necessary in any case to deal with the merge conflicts caused by #1239.

Reply:
Done.